src/psbt.cpp

```diff
@@ -467,36 +467,41 @@ PSBTError SignPSBTInput(const SigningProvider& provider, PartiallySignedTransact
     return sig_complete ? PSBTError::OK : PSBTError::INCOMPLETE;
```

Comment:
Can this commit be a separate PR? It would be nice to have a functional test that checks non witness utxos are dropped if sighash type is not `SIGHASH_ANYONECANPAY`.  It can be added in this PR as well at the cost of a larger diff.

Reply:
No, this was added because without it there were test failures.